modules/ztp-image-based-upgrade.adoc

```diff
@@ -0,0 +1,80 @@
+// Module included in the following assemblies:
+// Epic TELCOSTRAT-160 (4.15/4.16), story TELCODOCS-1576
+// * scalability_and_performance/ztp-talm-updating-managed-policies.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="ztp-image-based-upgrade-concept_{context}"]
+= Image-based upgrade for {sno} clusters with the {lcao}
+
+You can upgrade the platform version of the managed {sno} clusters with the {lcao}.
+By using generated a seed image, this upgrade process allows you to skip multiple minor releases and it is faster than the standard upgrade method.
+
+You generate the seed image from a cluster that is upgraded and configured with the desired {product-title} and applications by using the standard upgrade method. This is the seed cluster.
+
+[IMPORTANT]
+====
+The image-based upgrade uses custom images that are specific to the hardware platform the clusters are running on.
+Each different hardware platform requires a separate custom image.
+====
+
+With this seed image, you can upgrade the platform version on any {sno} that matches the image's cluster profile and then restore applications.
+
+//How do we specify the hardware exactly? Is it baked into the seed image based on the seed cluster we're using?
+
+The {lcao} uses two custom resources (CRs):
+
+* The `SeedGenerator` CR specifies the platform version of the seed cluster that you want to use to upgrade the target clusters.
+* The `imageBasedUpgrade` CR specifies the generated seed image's location and you use it to step through the upgrade stages.
+
+After generating the seed image, the upgrade process has four stages:
+
+* `Idle`
+* `Prep`
+* `Upgrade`
+* `Rollback` (Optional)
+
+[discrete]
+== Idle stage
+
+There is no ongoing upgrade and the cluster is ready to move to the `Prep` stage.
+
+[discrete]
+== Prep stage
+
+You can complete this stage in advance.
+
+The {lcao} ensures that the node is ready to proceed to the `Upgrade` stage. This includes the following steps:
+
+* Installing and configuring the OADP Operator
+* Setting up the storage bucket for backup
+* Precaching list of images from the seed image
+
+The {lcao} uses the `ostree` versioning system and the OADP Operator to create a backup, which allow complete cluster reconfiguration after upgrade and rollback.
+
+At this stage, the {lcao} pulls the seed image to the target cluster and two stateroots are created by `ostree`:
+
+* stateroot of the current version of the target cluster
+* stateroot of the newer version that is based on the seed image
+
+You can stop the upgrade process at this stage. In that case, the Operator performs cleanup operations and sets the upgrade stage back to `Idle`.
+
+[discrete]
+== Upgrade stage
+
+Just before the {lcao} starts the upgrade process, the OADP Operator creates a backup of the artifacts on a compatible storage solution, such as an Amazon S3 bucket. These OADP `Backup` and `Restore` CRs are also copied to the new stateroot to be applied after the upgrade
```

Comment:
I would encourage to add as an example Ceph Object Gateway which is compatible with Amazon S3. In my case I've been using MinIO which is an OCP certified operator IIRC.

Reply:
We should not be dictating the backend